dbflute-engine/src/main/java/org/dbflute/logic/doc/spolicy/DfSPolicyColumnStatementChecker.java

```diff
@@ -175,8 +183,8 @@ protected void evaluateColumnThenTheme(DfSPolicyStatement statement, DfSPolicyRe
         final String thenTheme = thenClause.getThenTheme(); // already not null here
         final boolean notThenClause = thenClause.isNotThenTheme();
         final String notOr = notThenClause ? "not " : "";
-        if (thenTheme.equalsIgnoreCase("bad") == !notThenClause) {
-            result.violate(policy, "The column is no good: " + toColumnDisp(column));
+        if (thenTheme.equalsIgnoreCase("bad") && notThenClause) {
```

Comment:
Does it work when "if columnName is suffix:_FLAG then bad" ?

Reply:
Removed the bug.
Additionally, I found and fixed a similar code for checking 'bad' theme of tableMap.
The previous code did not check 'then not' theme properly as well.